core/graph/src/main/java/uk/gov/gchq/gaffer/graph/Graph.java

```diff
@@ -988,13 +985,13 @@ private void initStore(final GraphConfig config) {
                 }
             }
 
-            // Use the store's graph Id if we don't have on configured already
+            // Use the store's graph Id if we don't have one configured already
             if (config.getGraphId() == null) {
                 config.setGraphId(store.getGraphId());
             }
 
             // Use the store's schema if not already set
-            if (schema == null || schema.getGroups().isEmpty()) {
+            if (schema == null) {
```

Comment:
Interesting that your fix removes checks, will this not remove crashes with incorrect config?
I was thinking `if (getSchema() == null || schema.getGroups().isEmpty())` might work, did you try that?

Reply:
> will this not remove crashes with incorrect config?

It shouldn't cause a problem because other checks are already made if the schema is non-null. These will already reject an empty schema is that is not valid for the store type. So I think this line would only have been reached when using a schemaless store.

For your other comment, [see my response in the issue](https://github.com/gchq/Gaffer/issues/3129#issuecomment-1863071145).